src/xava.c

```diff
@@ -51,24 +51,36 @@ static XAVA xava;
 static pthread_t p_thread;
 
 void handle_ionotify_call(XAVA_IONOTIFY_EVENT event, const char *filename,
-        int id, XAVA *xava) {
-    UNUSED(id);
-    UNUSED(xava);
+        int id, XAVA *xavaionot) {
     UNUSED(filename);
-    switch(event) {
-        case XAVA_IONOTIFY_CHANGED:
-        //case XAVA_IONOTIFY_DELETED: // ignore because it is just broken
-            should_reload = 1;
-            break;
-        case XAVA_IONOTIFY_ERROR:
-            xavaBail("ionotify event errored out! Bailing...");
+    UNUSED(xavaionot);
+    switch(id) {
+        case XAVA_IONOTIFY_CALLBACK_MAIN:
+            printf("triggered non pywal!\n");
+            switch(event) {
+                case XAVA_IONOTIFY_CHANGED:
+                //case XAVA_IONOTIFY_DELETED: // ignore because it is just broken
+                    should_reload = 1;
+                    break;
+                case XAVA_IONOTIFY_ERROR:
+                    xavaBail("ionotify event errored out! Bailing...");
+                    break;
+                case XAVA_IONOTIFY_CLOSED:
+                    xavaLog("ionotify socket closed");
+                    break;
+                default:
+                    xavaLog("Unrecognized ionotify call occured!");
+                    break;
+            }
             break;
-        case XAVA_IONOTIFY_CLOSED:
-            xavaLog("ionotify socket closed");
+        case XAVA_IONOTIFY_CALLBACK_PYWAL:
+            printf("triggered pywal!\n");
+            XAVA_CONFIG     *p      = &xava.conf;
```

Comment:
This shouldn't be the correct approach, we should trigger a window redraw instead. You're pulling the global xava reference instead of the argument which is undefined behavior.

To redraw the window move the redrawWindow if block from the main loop into its own function, call it ``xava_window_redraw`` (just to have a consistent naming scheme) and call it with the xava argument from the callback.